test/functional/pdo_sqlsrv/test_encoding_UTF8_emulate_prepare.phpt

```diff
@@ -1,113 +1,86 @@
-	}
-	$outValue1 = null;
-	
-	$value1 = $stmt4->fetchcolumn(1);
-	$outValue1 = $value1;
-	if (strncmp($inValue1, $outValue1, $inValueLen) == 0) {
-		echo "outValue is the same as inValue.\n";
-	}
-	$outvalue1 = null;
-	
-	$value2 = $stmt4->fetchColumn(1);
-	$outValue1 = $value2;
-	if (strncmp($inValue1, $outValue1, $inValueLen) == 0) {
-		echo "outValue is the same as inValue.\n";
-	}
-	$outValue1 = null;
-	
-	$value3 = $stmt4->fetchColumn(1);
-	$outValue1 = $value3;
-	if (strncmp($inValue1, $outValue1, $inValueLen) == 0) {
-		echo "outValue is the same as inValue.\n";
-	}
-	$outValue1 = null;
-	
-	$value4 = $stmt4->fetchColumn(1);
-	$outValue1 = $value4;
-	if (strncmp($inValue1, $outValue1, $inValueLen) == 0) {
-		echo "outValue is the same as inValue.\n";
-	}
-	$stmt4 = null;
-	
-	$stmt5 = $conn->prepare( "SELECT ? = c2_char FROM [Table_UTF]", array(PDO::ATTR_EMULATE_PREPARES => true) );	
-	$stmt5->setAttribute(constant('PDO::SQLSRV_ATTR_ENCODING'), PDO::SQLSRV_ENCODING_UTF8);
-	$outValue1 = "hello";	
-	$stmt5->bindParam( 1, $outValue1, PDO::PARAM_STR, 1024);
-	$stmt5->execute();	
-	if (strncmp($inValue1, $outValue1, $inValueLen) == 0) {
-		echo "outValue is the same as inValue.\n";
-	} else {
-		echo "outValue is $outValue1\n";
-	}
-	$stmt5 = null;
+    $stmt3 = $conn->prepare("INSERT INTO [Table_UTF] (c1_int, c2_char) VALUES (:var1, :var2)");
+    $stmt3->setAttribute(PDO::SQLSRV_ATTR_ENCODING, PDO::SQLSRV_ENCODING_UTF8);
+    $stmt3->bindParam(2, $inValue1);
+    $stmt3->bindValue(1, 1);
+    $stmt3->execute();
+    $stmt3->bindValue(1, 2);
+    $stmt3->execute();
+    $stmt3->bindValue(1, 3);
+    $stmt3->execute();
+    $stmt3->bindValue(1, 4);
+    $stmt3->execute();
+    $stmt3->bindValue(1, 5);
+    $stmt3->execute();
+    unset($stmt3);
 
-	$stmt6 = $conn->query("DROP TABLE [Table_UTF]");	
-	$stmt6 = null;
-	
-	$conn = null;	
-}
-catch (PDOexception $e){
-    print_r( ($e->errorInfo)[2] );
+    $stmt4 = $conn->prepare("SELECT * FROM $tbname");
+    $stmt4->setAttribute(PDO::SQLSRV_ATTR_ENCODING, PDO::SQLSRV_ENCODING_UTF8);
+    $outValue1 = null;
+    $stmt4->execute();
+    $row1 = $stmt4->fetch(PDO::FETCH_NUM);
+    $count1 = count($row1);
+    echo("Number of columns: $count1\n");
+    $v0 = $row1[0];
+    $outValue1 = $row1[1];
+    if (strncmp($inValue1, $outValue1, $inValueLen) == 0) {
+        echo "outValue is the same as inValue.\n";
+    }
+
+    for ($i = 0; $i < 4; $i++) {
+        $outValue1 = $stmt4->fetchColumn(1);
+        if (strncmp($inValue1, $outValue1, $inValueLen) == 0) {
+            echo "outValue is the same as inValue.\n";
+        }
+    }
+    unset($stmt4);
+
+    $option;
+    if (!isAEConnected()) {
+        $option[PDO::ATTR_EMULATE_PREPARES] = true;
+    } else {
+        $option[PDO::ATTR_EMULATE_PREPARES] = false;
+    }
+    $stmt5 = $conn->prepare("SELECT ? = c2_char FROM $tbname", $option);
+    $stmt5->setAttribute(PDO::SQLSRV_ATTR_ENCODING, PDO::SQLSRV_ENCODING_UTF8);
+    $outValue1 = "hello";
```

Comment:
setting encoding to UTF-8 is redundant... It's the default, isn't it?

Reply:
the connection attribute was set to ENCODING_SYSTEM, and this ones sets the statement attribute to UTF8